### PR TITLE
Apple M1 support

### PR DESCRIPTION
**Pull request by fhunleth**
All Homebrew blockers have been fixed. I think this works. So far the armv7 toolchain built completely so I'm optimistic that this will either work for all toolchains or the remaining issues will be minor!

The commits to be merged are the 3 substantive ones. I plan to merge manually on approval and skip merging the dependency commits.

### CODE REVIEW

File: nerves_toolchain_ctng/build.sh [hunk 1]
```diff
@@ -122,21 +125,23 @@ if [[ $BUILD_OS = "darwin" ]]; then
...
-    CROSSTOOL_LDFLAGS="-L/usr/local/opt/gettext/lib -lintl"
-    CROSSTOOL_CFLAGS="-I/usr/local/opt/gettext/include"
+    export CURSES_LIBS="-L$(brew --prefix ncurses)/lib -lncursesw"
+    CROSSTOOL_LDFLAGS="-L$HOMEBREW_PREFIX/lib -lintl"
+    CROSSTOOL_CFLAGS="-I$HOMEBREW_PREFIX/include"
 
     # Apple provides an old version of Bison that will fail about 20 minutes into the build.
-    export PATH="/usr/local/opt/bison/bin:$PATH"
-    if [[ ! -e /usr/local/opt/bison/bin/bison ]]; then
+    export PATH="$(brew --prefix bison)/bin:$PATH"
```
Comment (jjcarstens) [inline nerves_toolchain_ctng/build.sh:133]: Are bison and grep going to be in a nonstandard spot which is why HOMEBREW_PREFIX won't work here and it needs to be called again?

Comment (fhunleth) [reply nerves_toolchain_ctng/build.sh:133]: Don't know. The call is free compared to the time it takes to get errors, so I didn't dig into it. I'm quite sure the script could be simplified more.

**Review by mobileoverlord — APPROVED** [review]
This is pretty cool. 🚀

File: nerves_toolchain_ctng/build.sh (hunk 1 above)
Comment (fhunleth) [reply nerves_toolchain_ctng/build.sh:133]: @jjcarstens I feel totally guilty about not wanting to change anything now that it works, and being 99.999% sure that the change that you suggest will totally work. I going to merge, but I know that every time I see this I'll have the urge to clean it up. The next time that I have time and energy, I'll do that.

**Pull request merged**